lib/mayaUsd/render/vp2RenderDelegate/draw_item.cpp

```diff
@@ -54,4 +51,20 @@ HdVP2DrawItem::~HdVP2DrawItem()
     }
 }
 
+void HdVP2DrawItem::AddRenderItem(MHWRender::MRenderItem* item, const HdGeomSubset* geomSubset)
+{
+    _renderItems.push_back(RenderItemData());
+    RenderItemData& renderItemData = _renderItems.back();
+
+    renderItemData._renderItem = item;
+    renderItemData._renderItemName = _drawItemName;
+    if (geomSubset) {
+        renderItemData._geomSubset = *geomSubset;
+        renderItemData._renderItemName += geomSubset->id.GetString().c_str();
```

Comment:
We are combining two SdfPaths in a string. Do we need a separator to make it readable and easier for debugging?

Reply:
FWIW, GetText() or data() serves the same purpose of GetString().c_str().

Reply:
Did you test selection on geom subsets? We rely on the render item name to find the index path so I think we need to update the selection interpretation code in `ProxyRenderDelegate::getInstancedSelectionPath` so we can extract the index path from the new name properly?

Reply:
Right now the output looks like:

`/Proxy_Fender_stratocaster_usdShape_0000023131BC8120/fender_stratocaster/StratocasterStand/DrawItem_000002314B068E60/fender_stratocaster/StratocasterStand/pxrUsdPreviewSurface10SG
`

The material path starts with a "/" so it ends up looking reasonable. We could add another character in between the object and material to make it clearer.

I thought selection was working but looking more carefully it is simply selecting the top level proxy shape, not the specific rprim. The code in proxyRenderDelegate is failing to convert the name into a valid rprim id as you have guessed.

I think what I will do is modify the render item name so that it starts with the rprimId, then include a separator which is not USD_UFE_SEPARATOR, then include extra information like instancing information or geom subset material id. That will make it simple to find that unique separator in getInstancedSelectionPath so we can select things correctly.

Reply:
Make sure to run interactive tests to confirm everything works for the recently contributed point instancer selection.

Reply:
point instancer selection is working as long as I have the changes for https://github.com/Autodesk/maya-usd/pull/1186 in place, so wait for that to get merged before this one.

Reply:
I'm assuming this is now outdated information. Resolving this conversation.